test/src/unit-capi-dense_array.cc

```diff
@@ -3525,6 +3538,19 @@ TEST_CASE_METHOD(
   CHECK(a1_buffer[0] == 13);
 
   // Reopen the array to see the new fragment
+  tiledb_config_free(&cfg);
+  err = nullptr;
+  REQUIRE(tiledb_config_alloc(&cfg, &err) == TILEDB_OK);
+  REQUIRE(err == nullptr);
+  rc = tiledb_config_set(
+      cfg,
+      "sm.array.timestamp_end",
+      std::to_string(tiledb::sm::utils::time::timestamp_now_ms()).c_str(),
+      &err);
+  REQUIRE(rc == TILEDB_OK);
+  REQUIRE(err == nullptr);
+  rc = tiledb_array_set_config(ctx_, array, cfg);
```

Comment:
This is a good catch for using `reopen`. I think we need to document this though as "breaking behavior" in HISTORY. 

Currently, `reopen` opens the array at the _current_ timestamp, regardless of the invocation `open_at` you might have used before. Here, the user sets the array config with an end timestamp and that is used also when reopening the array. So, we correctly reset the config before we reopen the array in this PR. Unfortunately, this may not be clear for the user who may just attempt to replace all `open_at` commands with `set_config + open`, i.e., they will not think twice to add a new `set_config` command before their `reopen` commands.